--- a/bin/charmc	2020-08-05 01:13:21.000000000 -0400
+++ b/bin/charmc	2020-08-20 20:04:29.558948765 -0400
@@ -434,14 +434,10 @@
 	PROG_EXT=".exe"
 fi
 
 CHARMDIR="$(getAbsPath "$CHARMBIN/..")"
-CHARMLIB="$CHARMDIR/lib"
-CHARMINC="$CHARMDIR/include"
-CHARMLIBSO=
-if test -d "$CHARMDIR/lib_so"
-then
-  CHARMLIBSO="$CHARMDIR/lib_so"
-fi
+CHARMLIB="gentoo-libdir"
+CHARMINC="/usr/include/gentoo-include"
+CHARMLIBSO="gentoo-libsodir"
 
 ##############################################################################
 #
